### PR TITLE
HW React2 Week1

### PR DESCRIPTION
**Pull request by slastushevskyi**

### CODE REVIEW

File: react/react2/week1/vite-project/src/UserSearchComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const [value, setValue] = useState("");
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    // To prevent showing latest result
+    setUsers([]);
+    // I created func with short delay to fetch data only when user stopped typing
```
Comment (LucasIversen) [inline react/react2/week1/vite-project/src/UserSearchComponent.jsx:16]: Smart!!

File: react/react2/week1/vite-project/src/UserSearchComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }, [value]);
+  return (
+    <>
+      <h1>Github user searcher</h1>
+      <h2>{value}</h2>
+      <UserInput value={value} setValue={setValue} />
+      {value.length > 0 && (
+        <ul>
+          {loading && <p>Loading...</p>}
+          {!loading && users.map((user) => <li key={user.id}>{user.login}</li>)}
```
Comment (LucasIversen) [inline react/react2/week1/vite-project/src/UserSearchComponent.jsx]: List item could have been put in a seperate component, but it's also fine like this when it's so small

**Review by LucasIversen — APPROVED** [review]
Hey Nikita.. Good job 👍

File: react/react2/week1/vite-project/src/UserSearchComponent.jsx (hunk 2 above)
Comment (slastushevskyi) [reply react/react2/week1/vite-project/src/UserSearchComponent.jsx]: Yes, I'll create list component for that. Need more practise with props :)

**Pull request merged**